tools/python/youtube_info_updater/youtube_info_updater.py

```diff
@@ -0,0 +1,171 @@
+    
+    def set_api_url(self):
+        """
+        Sets the api url.
+        """
+        self.__google_api_url = f'{self.__google_api_url}&key={self.__youtube_api_key}'
+
+    def set_leagues(self):
+        """
+        Sets the league dictionary.
+        """
+        self.__leagues = {
+            'shl': self.__shl_channel_id,
+            'smjhl': self.__smjhl_channel_id,
+            'wjc': self.__wjc_channel_id,
+            'iihf': self.__iihf_channel_id
+        }
+
+# database functions
+
+    def set_database_information(self):
+        """
+        Sets the data base information from a .env.local file in the path given.
+        """
+        config = dotenv_values(self.__env_file_path)
+        self.__mysql_host = config['MYSQL_HOST']
+        self.__mysql_database = config['MYSQL_DATABASE']
+        self.__mysql_user = config['MYSQL_USER']
+        self.__mysql_password = config['MYSQL_PASSWORD']
+        self.__youtube_api_key = config['NEXT_PUBLIC_YOUTUBE_API_KEY']
+        self.__shl_channel_id = config['NEXT_PUBLIC_SHL_CHANNEL_ID']
+        self.__smjhl_channel_id = config['NEXT_PUBLIC_SMJHL_CHANNEL_ID']
+        self.__wjc_channel_id = config['NEXT_PUBLIC_SMJHL_CHANNEL_ID']
+        self.__iihf_channel_id = config['NEXT_PUBLIC_SMJHL_CHANNEL_ID']
+
+    def connect_to_database(self):
+        """
+        Connect to the database.
+        """
+        self.__mysql_connection = mysql.connector.connect(
+            host=self.__mysql_host,
+            user=self.__mysql_user,
+            password=self.__mysql_password,
+            database=self.__mysql_database
+        )
+
+    def select_all(self):
+        """
+        Select all from the database table.
+        """
+        cursor = self.__mysql_connection.cursor()
+        cursor.execute('SELECT * FROM youtube_data')
+        result = cursor.fetchall()
+        return result
+
+    def update_league(self, league, video_info):
+        """
+        Update a league in the database to a new video id.
+        """
+        # separate the tuple to get the individual elements
+        video_id = video_info[0]
+        video_is_live = video_info[1]
+
+        cursor = self.__mysql_connection.cursor()
+        statement = f"UPDATE youtube_data "
+        statement = statement + f"SET videoID = '{video_id}', isLive = '{video_is_live}' "
+        statement = statement + f"WHERE league = '{league.lower()}'"
+        cursor.execute(statement)
+
+        self.__mysql_connection.commit()
+
+        print(f'league [{league}] updated to video id [{video_id}] if it exists.')
+
+# setters/getters
+
+    def set_env_file_path(self, env_file_path):
+        """
+        Sets the env_file_path.
+        """
+        self.__env_file_path = env_file_path
+
+    def get_env_file_path(self):
+        """
+        Returns the env_file_path.
+        """
+        return self.__env_file_path
+
+# testing functions
+
+    def run_test(self, *args):
```

Comment:
a function that does nothing.  Kept so that we can test any functionalities later.